### PR TITLE
switch to the default implementation of `write_vectored`

### PR DESCRIPTION
**Pull request by stlankes**
HermitOS doesn't support write_vectored and switch to the default implementation of `write_vectored`.

### CODE REVIEW

File: library/std/src/sys/pal/hermit/net.rs [hunk 1]
```diff
@@ -225,17 +225,11 @@ impl Socket {
...
 
     pub fn write_vectored(&self, bufs: &[IoSlice<'_>]) -> io::Result<usize> {
-        let mut size: isize = 0;
-
-        for i in bufs.iter() {
-            size += cvt(unsafe { netc::write(self.0.as_raw_fd(), i.as_ptr(), i.len()) })?;
-        }
-
-        Ok(size.try_into().unwrap())
+        crate::io::default_write_vectored(|b| self.write(b), bufs)
```
Comment (workingjubilee) [inline library/std/src/sys/pal/hermit/net.rs:228]: huh, this appears to be the exact same implementation as what's on the left. weird.

Comment (stlankes) [reply library/std/src/sys/pal/hermit/net.rs:228]: Yes, but is_write_vectored was wrong.

Comment (workingjubilee) [reply library/std/src/sys/pal/hermit/net.rs:228]: yep!

**Comment by workingjubilee** [comment]
lgtm
@bors r+

File: library/std/src/sys/pal/hermit/net.rs (hunk 1 above)
Comment (tbu-) [reply library/std/src/sys/pal/hermit/net.rs:228]: You could probably just delete the implementation altogether, it'll fall back to the default automatically.

Comment (workingjubilee) [reply library/std/src/sys/pal/hermit/net.rs:228]: This is the impl of Socket, tho, innit?

Or do you mean that the `write_vectored` impls will resolve to the trait and then dispatch to that one's default impl?

**Comment by workingjubilee** [comment]
Wait.

@bors r-

@stlankes I think the `read_vectored` implementation is also wrong in the same way, looking at it.

File: library/std/src/sys/pal/hermit/net.rs (hunk 1 above)
Comment (tbu-) [reply library/std/src/sys/pal/hermit/net.rs:228]: > This is the impl of Socket, tho, innit?

Oh yeah. Nevermind my comment then, didn't notice.

**Comment by stlankes** [comment]
> Wait.
> 
> @bors r-
> 
> @stlankes I think the `read_vectored` implementation is also wrong in the same way, looking at it.

Good point! I oversaw it.

**Comment by workingjubilee** [comment]
Nice, okay!
@bors r+ rollup

**Pull request merged**